<a href="https://colab.research.google.com/github/bosywahab818-a11y/flyrank-ml-internship-bouthina/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/bosywahab818-a11y/flyrank-ml-internship-bouthina/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My Lane

I chose the **Refresh / Content Opportunity Scoring** lane.

I chose this lane because it focuses on identifying pages that should be reviewed first before they lose more traffic. This problem has a clear business value because content teams have limited time and need to prioritize which pages to update first. The starter dataset already contains useful features such as impressions, CTR, rankings, content age, and trend direction that can help support this decision.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# The Question

## Decision

The decision is to identify which content pages should be reviewed and refreshed first.

## Action

The SEO or content team can use this ranked list to prioritize updating pages that are more likely to decline in performance.

## Cost of a Wrong Recommendation

If a healthy page is recommended, the team may waste time and resources reviewing it.

If a declining page is missed, it may continue losing traffic and visibility, reducing the website's overall performance.

## Why Machine Learning?

Machine learning can analyze many features at the same time, making it easier to prioritize pages than using simple manual rules alone.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
import os
import pandas as pd

# Clone the repository if it doesn't exist
repo_name = "flyrank-ml-internship-bouthina"
if not os.path.exists(repo_name):
    !git clone https://github.com/bosywahab818-a11y/flyrank-ml-internship-bouthina.git

# Change the current working directory to the repository root
# This allows the original relative path to work
# We save the original directory to return to it later if needed, but for this cell, it's not strictly necessary
original_cwd = os.getcwd()
os.chdir(repo_name)

# Load the starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

# 1. Number of pages
print("Dataset size:", len(df), "pages")

# 2. Percentage of declining pages
declining_rate = (df["trend_direction"] == "down").mean() * 100
print(f"Declining pages: {declining_rate:.2f}%")

# 3. Correlation between search volume and impressions
corr = df["search_volume"].corr(df["impressions_90d"])
print(f"Correlation between search volume and impressions: {corr:.3f}")

# Restore original working directory (optional, but good practice if subsequent cells depend on it)
os.chdir(original_cwd)

# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

Dataset size: 30000 pages
Declining pages: 54.21%
Correlation between search volume and impressions: 0.001


# Quick Look at the Data

The starter dataset contains **30,000** content pages.

About **54.2%** of the pages are labeled as declining, showing that content decline is a common problem worth investigating.

The correlation between **search volume** and **impressions** is approximately **0.001**, which suggests that keyword search volume alone is not enough to explain how much traffic a page actually receives. This supports the need for a machine learning approach that considers multiple features together.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

# Careful Words

This project is a decision-support system. It aims to identify content pages that may deserve review based on observed patterns in the data.

The results can show associations and help prioritize pages for review, but they do not prove that refreshing a page will improve its performance.

This analysis does not predict Google's algorithm or prove cause-and-effect relationships. Any recommendations should be reviewed by human experts before taking action.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.